apps/vmq_enhanced_auth/src/vmq_enhanced_auth.erl

```diff
@@ -954,4 +966,139 @@ simple_acl(_) ->
             split_topic_label(<<"a/b label a b\n">>)
         )
     ].
+add_complex_acl_test(_) ->
+    ACL = [<<"topic abc/xyz/#\n">>],
+    load_from_list(ACL),
+    Topic1 = [<<"abc">>, <<"xyz">>, <<"+">>, <<"1">>],
+    Topic2 = [<<"abc">>, <<"xyz">>, <<"+">>, <<"2">>],
+    vmq_reg_redis_trie:add_complex_topic("", Topic1),
+    vmq_reg_redis_trie:add_complex_topic("", Topic2),
+    [
+        ?_assertEqual(
+            {ok, [
+                {Topic1, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic1, 0}
+                ]
+            )
+        ),
+        ?_assertEqual(
+            {ok, [
+                {Topic2, 0, {matched_acl, undefined, <<"abc/xyz/#">>}}
+            ]},
+            auth_on_subscribe(
+                <<"test">>,
+                {"", <<"my-client-id">>},
+                [
+                    {Topic2, 0}
+                ]
+            )
+        )
+    ].
+delete_complex_acl_test(_) ->
```

Comment:
These tests are for complex trie which is not implemented in vmq_enhanced_auth plugin.

Reply:
Removed